### ISSUE TITLE
Negative samples/images

### ISSUE THREAD

**Issue by vasnakh** [ultralytics/yolov5#3218]
## ❔Question
Was wondering if it is possible to include images that have no bounding boxes to help the object detector have less false positives? I am seeing many false positives and I believe that's because the model hasn't seen the backgrounds well enough. Note that the image size of the original image is large and there are few objects in the image so I applied tiling and basically the training tiles are all around the object itself not on the edges of the actual/large image. Any suggestions/helps is highly appreciated.

**Comment by wudashuo** [comment]
Yes, you can add some negative images. Simply put images with no label or empty label in your dataset.

**Comment by vasnakh** [comment]
Thank you. It was a bit confusing since I saw the following being printed but now I understand that this is just about the labels. 

train: Scanning 'data/dataset/sar_dataset/labels/train' images and labels... 1593 found, 12737 missing, 0 empty, 0 corrupted

**Comment by glenn-jocher** [comment]
@vasnakh yes it's fine to have missing labels. In your example it looks like you've included many background images in relation to your training images. Generally we recommend a number around 0-10%, though you can raise this higher to reduce FPs. In your dataset the ratio is about 800%, so you may want to either add more training images or reduce your backgrounds a bit. Full training recommendations are below.

👋 Hello! Thanks for asking about improving training results. Most of the time good results can be obtained with no changes to the models or training settings, **provided your dataset is sufficiently large and well labelled**. If at first you don't get good results, there are steps you might be able to take to improve, but we always recommend users **first train with all default settings** before considering any changes. This helps establish a performance baseline and spot areas for improvement.

If you have questions about your training results **we recommend you provide the maximum amount of information possible** if you expect a helpful response, including results plots (train losses, val losses, P, R, mAP), PR curve, confusion matrix, training mosaics, test results and dataset statistics images such as labels.png. All of these are located in your `project/name` directory, typically `yolov5/runs/train/exp`.

We've put together a full guide for users looking to get the best results on their YOLOv5 trainings below.

## Dataset

- **Images per class.** ≥1.5k images per class
- **Instances per class.** ≥10k instances (labeled objects) per class total
- **Image variety.** Must be representative of deployed environment. For real-world use cases we recommend images from different times of day, different seasons, different weather, different lighting, different angles, different sources (scraped online, collected locally, different cameras) etc.
- **Label consistency.** All instances of all classes in all images must be labelled. Partial labelling will not work.
- **Label accuracy.** Labels must closely enclose each object. No space should exist between an object and it's bounding box. No objects should be missing a label.
- **Background images.** Background images are images with no objects that are added to a dataset to reduce False Positives (FP). We recommend about 0-10% background images to help reduce FPs (COCO has 1000 background images for reference, 1% of the total).

<a href="https://arxiv.org/abs/1405.0312"><img width="800" src="https://user-images.githubusercontent.com/26833433/109398377-82b0ac00-78f1-11eb-9c76-cc7820669d0d.png" alt="COCO Analysis"></a>


## Model Selection

Larger models like YOLOv5x and [YOLOv5x6](https://github.com/ultralytics/yolov5/releases/tag/v5.0) will produce better results in nearly all cases, but have more parameters, require more CUDA memory to train, and are slower to run. For **mobile** deployments we recommend YOLOv5s/m, for **cloud** deployments we recommend YOLOv5l/x. See our README [table](https://github.com/ultralytics/yolov5#pretrained-checkpoints) for a full comparison of all models. 

<p align="center"><img width="700" alt="YOLOv5 Models" src="https://user-images.githubusercontent.com/26833433/103595982-ab986000-4eb1-11eb-8c57-4726261b0a88.png"></p>

- **Start from Pretrained weights.** Recommended for small to medium sized datasets (i.e. VOC, VisDrone, GlobalWheat). Pass the name of the model to the `--weights` argument. Models download automatically from the [latest YOLOv5 release](https://github.com/ultralytics/yolov5/releases).
```bash
python train.py --data custom.yaml --weights yolov5s.pt
                                             yolov5m.pt
                                             yolov5l.pt
                                             yolov5x.pt
```
- **Start from Scratch.** Recommended for large datasets (i.e. COCO, Objects365, OIv6). Pass the model architecture yaml you are interested in, along with an empty `--weights ''` argument:
```bash
python train.py --data custom.yaml --weights '' --cfg yolov5s.yaml
                                                      yolov5m.yaml
                                                      yolov5l.yaml
                                                      yolov5x.yaml
```


## Training Settings

Before modifying anything, **first train with default settings to establish a performance baseline**. A full list of train.py settings can be found in the [train.py](https://github.com/ultralytics/yolov5/blob/master/train.py) argparser.

- **Epochs.** Start with 300 epochs. If this overfits early then you can reduce epochs. If overfitting does not occur after 300 epochs, train longer, i.e. 600, 1200 etc epochs.
- **Image size.** COCO trains at native resolution of `--img 640`, though due to the high amount of small objects in the dataset it can benefit from training at higher resolutions such as `--img 1280`. If there are many small objects then custom datasets will benefit from training at native or higher resolution. Best inference results are obtained at the same `--img` as the training was run at, i.e. if you train at `--img 1280` you should also test and detect at `--img 1280`.
- **Batch size.** Use the largest `--batch-size` that your hardware allows for. Small batch sizes produce poor batchnorm statistics and should be avoided.
- **Hyperparameters.** Default hyperparameters are in [hyp.scratch.yaml](https://github.com/ultralytics/yolov5/blob/master/data/hyp.scratch.yaml). We recommend you train with default hyperparameters first before thinking of modifying any. In general, increasing augmentation hyperparameters will reduce and delay overfitting, allowing for longer trainings and higher final mAP. Reduction in loss component gain hyperparameters like `hyp['obj']` will help reduce overfitting in those specific loss components. For an automated method of optimizing these hyperparameters, see our [Hyperparameter Evolution Tutorial](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution).

## Further Reading

If you'd like to know more a good place to start is Karpathy's 'Recipe for Training Neural Networks', which has great ideas for training that apply broadly across all ML domains:
http://karpathy.github.io/2019/04/25/recipe/

**Comment by jaqub-manuel** [comment]
Dear, @glenn-jocher 
Why do you usually recommend a number around 0-10% for negative samples? Is there a paper or experiment that supports what you say?
Many thanks in advance.

**Comment by glenn-jocher** [comment]
@jaqub-manuel higher background images reduce FPs at the expense of reduced TPs. Use as many or as few as you see fit for your use case.

**Comment by kbratsy** [comment]
Hello @glenn-jocher, I want to improve an already-trained model with negative images. Can I fine-tune the model by providing only the background images? Or do I need to retrain the model from scratch with labeled and background images?

**Comment by glenn-jocher** [comment]
@kbratsy yes, you can fine-tune the already-trained model by providing only background images. There is no need to retrain the model from scratch with labeled images. Including background images in the fine-tuning process can help reduce False Positives (FP) and improve the performance of your model. Make sure to provide a sufficient number of background images to ensure better results.

**Comment by kbratsy** [comment]
Hi @glenn-jocher, thank you for your information. I want to ask one more question. Does the number of epochs matter when fine-tuning the already-trained model in background images? Is a small epoch number like 1 or 10 enough? Or would it be better to increase the number of epochs (e.g. 100) to reduce FPs?

**Comment by glenn-jocher** [comment]
@kbratsy, glad to assist you! When fine-tuning an already-trained model with background images, the number of epochs does indeed matter. A small epoch number like 1 or 10 may not be sufficient to effectively fine-tune the model and reduce False Positives (FPs). It is generally recommended to increase the number of epochs, such as 100 or even more, to allow the model to better adapt to the background images and improve its performance. Keep in mind that the optimal number of epochs can vary depending on your specific dataset and training scenario, so you may need to experiment with different values to find the best results.
